tgui/packages/tgui/components/Dropdown.tsx

```diff
@@ -234,6 +237,104 @@ export class Dropdown extends Component<DropdownProps, DropdownState> {
     }
   }
 
+  getOptionValue(option) {
+    return typeof option === 'string' ? option : option.value;
+  }
+
+  getOptionsValues() {
+    const { options = [] } = this.props;
+
+    return options.map((option) => {
+      return this.getOptionValue(option);
+    });
+  }
+
+  getSelectedIndex() {
+    const selected = this.state.selected
+      ? this.state.selected
+      : this.props.selected;
+    const { options = [] } = this.props;
+
+    let selectedIndex;
+    options.forEach((option, index) => {
+      let value = this.getOptionValue(option);
+
+      if (value === selected) {
+        selectedIndex = index;
+      }
+    });
+
+    return selectedIndex;
+  }
+
+  hasSwitchToPrevious() {
+    const selectedIndex = this.getSelectedIndex();
+
+    if (selectedIndex === undefined) {
+      return false;
+    }
+
+    const previousIndex = parseInt(selectedIndex, 10) - 1;
```

Comment:
I'm confused, why do we need to parse this as an int?

Reply:
To be sure it will be a sum and not a concatenation

Reply:
Still confused, in what situation is this not already a number